File: order-service/src/main/java/com/example/orderservice/service/OrderService.java
package com.example.orderservice.service;

import com.example.orderservice.dto.OrderRequest;
import com.example.orderservice.dto.OrderResponse;
import com.example.orderservice.entity.Order;

import java.util.List;

public interface OrderService {
    OrderResponse createOrder(OrderRequest request);
    OrderResponse getOrderById(Long id);
    OrderResponse getOrderByNumber(String orderNumber);
    List<OrderResponse> getOrdersByUserId(Long userId);
    List<OrderResponse> getOrdersByRestaurantId(Long restaurantId);
    OrderResponse updateOrderStatus(Long id, Order.OrderStatus status);
    List<OrderResponse> getOrdersByStatus(Order.OrderStatus status);
    void cancelOrder(Long id);
}

File: frontend/src/services/orderService.js
import api from './api';

const orderService = {
  createOrder: (orderData) => {
    return api.post('/api/orders', orderData);
  },

  getMyOrders: () => {
    return api.get('/api/orders');
  },

  getOrderById: (id) => {
    return api.get(`/api/orders/${id}`);
  },

  trackOrder: (id) => {
    return api.get(`/api/orders/${id}/track`);
  },

  cancelOrder: (id) => {
    return api.put(`/api/orders/${id}/cancel`);
  },

  updateOrderStatus: (id, status) => {
    return api.put(`/api/orders/${id}/status`, { status });
  },
};

export default orderService;
